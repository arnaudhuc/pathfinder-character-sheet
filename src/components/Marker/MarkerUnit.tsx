/** @jsx jsx */
import { jsx } from '@emotion/core';
import { common } from '../../style/common';
import { useState } from 'react';
import { green } from '@material-ui/core/colors';

interface IMarkerUnit {
	index: number;
	marker: string;
}

export const MarkerUnit: React.FC<IMarkerUnit> = (props) => {
	const { index, marker } = props;

	const [isSelected, setSelected] = useState(false);

	const handleClick = () => {
		setSelected(!isSelected);
	};

	return (
		<div
			css={{
				border: `3px solid ${isSelected ? green.A200 : common.grey}`,
				width: '30px',
				height: '30px',
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'center',
				transform: 'rotate(45deg)',
				position: 'absolute',
				top: index % 2 ? '10px' : '0',
				left: `${index * 30}px`,
				color: common.greyDark,
			}}
			onClick={handleClick}
		>
			<span css={{ transform: 'rotate(-45deg)' }}>{marker}</span>
		</div>
	);
};
