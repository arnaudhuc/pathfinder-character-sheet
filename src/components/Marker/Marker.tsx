/** @jsx jsx */
import { jsx } from '@emotion/core';
import { MarkerUnit } from './MarkerUnit';
interface IMarker {
	markers: string[];
}

export const Marker: React.FC<IMarker> = (props) => {
	const { markers } = props;
	return (
		<div
			css={{
				position: 'relative',
			}}
		>
			{markers.map((marker, index) => (
				<MarkerUnit key={index} index={index} marker={marker} />
			))}
		</div>
	);
};
