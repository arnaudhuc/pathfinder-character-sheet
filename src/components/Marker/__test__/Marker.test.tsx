/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';
import { Marker } from '../Marker';

describe('Marker', () => {
	test('should render with default props', () => {
		const container = renderer.create(<Marker markers={['1']} />).toJSON();

		expect(container).toMatchSnapshot();
	});

	test('should render multiple marker', () => {
		const container = renderer
			.create(<Marker markers={['1', '2', '3', '4']} />)
			.toJSON();

		expect(container).toMatchSnapshot();
	});
});
