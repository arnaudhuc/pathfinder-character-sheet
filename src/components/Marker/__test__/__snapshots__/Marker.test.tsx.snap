// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should render multiple marker 1`] = `
<div
  className="css-1naiqne-Marker"
>
  <div
    className="css-11hdqe0-MarkerUnit"
    onClick={[Function]}
  >
    <span
      className="css-twv3ha-MarkerUnit"
    >
      1
    </span>
  </div>
  <div
    className="css-5mf5wp-MarkerUnit"
    onClick={[Function]}
  >
    <span
      className="css-twv3ha-MarkerUnit"
    >
      2
    </span>
  </div>
  <div
    className="css-1yafj3t-MarkerUnit"
    onClick={[Function]}
  >
    <span
      className="css-twv3ha-MarkerUnit"
    >
      3
    </span>
  </div>
  <div
    className="css-1qhu3xl-MarkerUnit"
    onClick={[Function]}
  >
    <span
      className="css-twv3ha-MarkerUnit"
    >
      4
    </span>
  </div>
</div>
`;

exports[`Marker should render with default props 1`] = `
<div
  className="css-1naiqne-Marker"
>
  <div
    className="css-11hdqe0-MarkerUnit"
    onClick={[Function]}
  >
    <span
      className="css-twv3ha-MarkerUnit"
    >
      1
    </span>
  </div>
</div>
`;
